modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -30,7 +30,7 @@ class Configuration(object):
 
     def __init__(self, host="{{{basePath}}}",
                  api_key=None, api_key_prefix=None,
-                 username="", password=""):
+                 username=None, password=None):
```

Comment:
If you make these default to None, then invoking
config. get_basic_auth_token() will raise
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Truth testing for empty string returns false, so it makes sense to leave these as empty strings, that way your `if self.username and self.password` will work, and we will not raise an exception when we run self.get_basic_auth_token

Reply:
My reasoning for having both None and empty string was a concern about backward compatibility. For example, if somebody is using the generated SDK to test a system, they would probably want to be able to test their API service with an empty user string, empty password, or empty access token. Also, potentially some SDK users are using the SDK in a unit test scenario where the access token is an empty string and the UT is accepting empty string as a valid token. I don't know enough about the usage to determine if it's really important to minimize any potential backward compatibility breakage. Because of these use cases, I thought it wouldn't be a good idea to omit the authorization attribute (e.g. query, cookie, header) just because a string is empty.

Reply:
True. What do you think about my other comment where I elaborate my reasoning about the "None" value? I just realized I would need to change the test, and I could also change config. get_basic_auth_token()

Reply:
I have also added some unit tests.